inc/ConfigMap.h

```diff
@@ -24,12 +25,54 @@
 #include <vector>
 
 namespace PCOE {
-    class ConfigMap : public std::unordered_map<std::string, std::vector<std::string>> {
+    class ConfigMap {
     public:
         ConfigMap() = default;
         ConfigMap(const std::string& filename);
         ConfigMap(const int argc, char* argv[]);
 
+        /**
+         * @brief       Gets string vector at given key
+         * @param       key     The key from which the vector is retrieved
+         * @return      The string vector corresponding to the given key
+         */
+        const std::vector<std::string>& getVector(const std::string& key) const;
+
+        double getDouble(const std::string& key) const;
+
+        std::uint64_t getUInt64(const std::string& key) const;
+
+        std::int64_t getInt64(const std::string& key) const;
+
+        std::uint32_t getUInt32(const std::string& key) const;
+
+        std::int32_t getInt32(const std::string& key) const;
+
+        const std::string& getString(const std::string& key) const;
+
+        void set(const std::string& key, const std::vector<std::string>& vector);
+
+        void set(const std::string& key, const std::initializer_list<std::string>& list);
```

Comment:
All of the set methods from here down should take their `value` by value rather than reference. Generally, any parameter that is "cheap" to copy should be passed by value. (Cheap is a matter of opinion, but I usually go with <= 4 * sizeof(void*)). `std::initializer_list` is specifically designed to be "cheap" (it's generally just 2 pointers), while the rest of the sets are primitives that are <= 1 pointer in size.